backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -102,111 +108,103 @@ private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
                 return;
             }
 
-            if (MissionRunQueueIsEmpty(MissionRunQueue(robot.Id)))
+            if (MissionScheduling.MissionRunQueueIsEmpty(MissionRunQueue(robot.Id)))
             {
                 _logger.LogInformation("The robot was changed to available but there are no mission runs in the queue to be scheduled");
                 return;
             }
 
-            var missionRun = MissionRunQueue(robot.Id).First(missionRun => missionRun.Robot.Id == robot.Id);
+            var missionRun = MissionRunQueue(robot.Id).FirstOrDefault(missionRun => missionRun.Robot.Id == robot.Id);
```

Comment:
Should this be handled in an else statement for the if statement below? This is only necessary if MissionQueueFrozen is false

Reply:
Ok so what happened when I tried this was the robot became available before the mission queue frozen was set to false and because isar only send one event with robot available, the mission queue didn't run the next mission. So the emergency status was cleared but the robot didn't start any mission from the queue.. So I think it's best to keep it this way, what do you think?

Reply:
This doesn't lead to problems if the mission queue is still frozen after clearing the emergency state when the robot gets available because no emergency state mission will be found so the robot will just keep the first mission in the queue and run it, that's why it works the way it has been implemented

Reply:
It's a timing issue and I don't see another way of guaranteeing that isar will ping the robot status as available always before the mission queue frozen has been updated to false

Reply:
Could you acquire the lock used to freeze the mission queue here? So that it will run as soon as the mission queue becomes unlocked? I haven't personally tried it, but hopefully these mutexes can be passed around.

Reply:
Hmm I now see that we do not use any mutexes in this process, which can cause problems regardless. I'll need to investigate this algorithm a bit more.

Reply:
Can we maybe trigger an event when the mission run queue becomes unfrozen? And then do this in that event handler if we have not already done so?